nixos/tests/postgresql-wal2json.nix

```diff
@@ -0,0 +1,53 @@
+{ system ? builtins.currentSystem
+, config ? { }
+, pkgs ? import ../.. { inherit system config; }
+, postgresql ? null }:
+
+let
+  makeTest = import ./make-test-python.nix;
+  makePostgresqlWal2jsonTest = subTestName: postgresqlPackage: let
+  in makeTest {
+    name = "postgresql-wal2json-${subTestName}";
+    meta.maintainers = with pkgs.lib.maintainers; [ euank ];
+
+    nodes.machine = {
+      services.postgresql = {
+        package = postgresqlPackage;
+        enable = true;
+        extraPlugins = with postgresqlPackage.pkgs; [ wal2json ];
+        settings = {
+          wal_level = "logical";
+          max_replication_slots = "10";
+          max_wal_senders = "10";
+        };
+      };
+    };
+
+    testScript = ''
+      machine.wait_for_unit("postgresql")
+      machine.succeed(
+          "cp ${./postgresql/wal2json/example2.sql} /tmp/example2.sql"
```

Comment:
Originally the filepath was included in test output, so I was copying it to a deterministic location.

I've since updated the psql commandline to include `-q` (`--quiet`), which removes that output, so I've updated it to remove this `cp`.

Thanks for catching that!